solr/solrj/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -103,6 +103,16 @@
   public static final String STATE_TIMESTAMP_PROP = "stateTimestamp";
   public static final String COLLECTIONS_ZKNODE = "/collections";
   public static final String LIVE_NODES_ZKNODE = "/live_nodes";
+
+  /**
+   * The following, node_roles and roles.json are for assigning roles to
+   * nodes. The node_roles is the preferred way (using -Dsolr.node.roles param),
+   * and roles.json is used by legacy ADDROLE API command.
+   * TODO: Deprecate and remove support for roles.json in an upcoming release.
```

Comment:
This TODO shouldn't be part of the javadoc, it should be a separate line comment on `ROLES`

Reply:
Fixed, thanks.